### PR TITLE
fix(instantsearch): keep algoliasearch-helper as external dependency during build

### PR DESCRIPTION
**Pull request by dhayab**
**Summary**

Since the integration of the algoliasearch-helper to the monorepo, it is locally linked to other packages that depend on it (most notably, instantsearch.js). This created an issue during the building of the UMD version of the package, where babel doesn't transpile the helper dependency because now it is locally linked thanks to lerna.

This PR adds a babel configuration option that forces the library to be transpiled.

**Result**

InstantSearch.js's UMD package now works correctly and doesn't throw an error at runtime.

[CR-3961](https://algolia.atlassian.net/browse/CR-3961)  
Closes https://github.com/algolia/instantsearch/issues/5763

[CR-3961]: https://algolia.atlassian.net/browse/CR-3961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bundlesize.config.json
Comment (dhayab) [inline bundlesize.config.json:1]: I find the 10kb bump to be slightly worrying, wdyt?

**Comment by dhayab** [comment]
I'm working on a follow-up PR that will run an UMD e-commerce example on e2e, so that we can detect issues at runtime with this specific use-case.

File: bundlesize.config.json
Comment (Haroenv) [reply bundlesize.config.json:1]: that's a huge difference, what's in the diff?

Comment (Haroenv) [reply bundlesize.config.json:1]: first notable change I see is _AsyncGenerator, that shouldn't be needed

Comment (Haroenv) [reply bundlesize.config.json:1]: I'm also seeing "_rollupPluginBabelHelpers" in this version that's not in last released one

File: packages/instantsearch.js/scripts/rollup/rollup.config.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const plugins = [
     rootMode: 'upward',
     exclude: /node_modules/,
     extensions: ['.js', '.ts', '.tsx'],
+    sourceType: 'unambiguous',
```
Comment (Haroenv) [inline packages/instantsearch.js/scripts/rollup/rollup.config.js:28]: I also wonder if the helper can be considered external? maybe that helps (although I'm not sure)

Comment (Haroenv) [inline packages/instantsearch.js/scripts/rollup/rollup.config.js:28]:
```suggestion
    exclude: /node_modules|algoliasearch-helper/,
    extensions: ['.js', '.ts', '.tsx'],
```

this works for me @dhayab, it leaves commonjs in place for the commonjs plugin, and doesn't add a bunch of babel plugins, essentially still using the helper as an external dependency.


I'm surprised to why the change is only done in InstantSearch.js, probably needed for React InstantSearch Hooks, React InstantSearch Core and Vue InstantSearch too right?

Comment (dhayab) [reply packages/instantsearch.js/scripts/rollup/rollup.config.js:28]: Awesome, that seems to work indeed! As for React InstantSearch and Vue, we might have the same issue, it's probably that it's less common to have UMD implementations of those, maybe?

**Review by Haroenv — APPROVED** [review]
